src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts

```diff
@@ -9,7 +12,48 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
       return ExpenditureStep.Review;
     case ExpenditureStatus.Locked:
       return ExpenditureStep.Funding;
+    case ExpenditureStatus.Finalized:
+      return ExpenditureStep.Payment;
     default:
       return ExpenditureStep.Create;
   }
 };
+
+/**
+ * Returns a boolean indicating whether the expenditure is fully funded,
+ * i.e. the balance of each token is greater than or equal to the sum of its payouts
+ */
+export const isExpenditureFullyFunded = (expenditure?: Expenditure | null) => {
+  if (!expenditure) {
+    return false;
+  }
+
+  if (!expenditure.balances) {
+    return false;
+  }
+
+  const amountsByToken = expenditure.slots.map((slot) => {
```

Comment:
nitpicky, but could we name this something like `slotAmountsByToken`? `amountsByToken` sounds like an object with token addresses as keys and amounts as values tbh :D